source/Time.cpp

```diff
@@ -198,9 +198,8 @@ RK2Scheme::Step(real& dt)
 	r[1] = r[0] + rd[0] * (0.5 * dt);
 	Update(0.5 * dt, 1);
 	// And so we can compute the new derivative and apply it
-	CalcStateDeriv(0);
-	r[0] = r[0] + rd[0] * dt;
-
+	CalcStateDeriv(1);
```

Comment:
This change is actually not needed. In RK2 you can overwrite rd[0] since it is not needed anymore. I do not know if it is better for readability, but in my opinion it is better as it was before. I am open to change my mind though

Reply:
Right, I didn't realize that. From a readability perspective this seemed like it was the Euler method to me which is why I changed it. Does it make a big difference in efficiency either way?

Reply:
Totally uncritical. I think it can be read just fine, but the fact that you got twisted might be a signal of pitfall. So you decide. Just let me know

Reply:
I think we should change it. Both Matt and I were confused by that.

Reply:
Perfect then